### ISSUE TITLE
librabbitmq 1.6.1 installed with pip not working

### ISSUE THREAD

**Issue by jimmywan** [celery/librabbitmq#86]
I can't seem to get librabbitmq working and would appreciate any pointers in the right direction. I've got a docker container with a working service relying upon celery and I updated it to install rabbitmq-c and librabbitmq (via pip), but it doesn't seem to be working. The library appears to build without problems and I've installed librabbitmq via pip, but when I try to import librabbitmq it doesn't work.

```
# python
Python 2.7.11 |Continuum Analytics, Inc.| (default, Dec  6 2015, 18:08:32)
[GCC 4.4.7 20120313 (Red Hat 4.4.7-1)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
Anaconda is brought to you by Continuum Analytics.
Please check out: http://continuum.io/thanks and https://anaconda.org
>>> import librabbitmq
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: No module named librabbitmq

# pip freeze | egrep 'amqp|librabbitmq|celery'
amqp==2.0.3
celery==3.1.23
librabbitmq==1.6.1
```

This is my Dockerfile snippet:

```
# Install rabbitmq-c
ENV RABBITMQ_C_VERSION=0.8.0
RUN DEBIAN_FRONTEND=noninteractive \
    wget https://github.com/alanxz/rabbitmq-c/archive/v${RABBITMQ_C_VERSION}.tar.gz \
    && tar xzf v${RABBITMQ_C_VERSION}.tar.gz \
    && apt-get update \
    && apt-get install -q -y cmake \
    && cd rabbitmq-c-${RABBITMQ_C_VERSION} \
    && mkdir build \
    && cd build \
    && cmake -DBUILD_EXAMPLES=OFF -DBUILD_TESTS=OFF -DENABLE_SSL_SUPPORT=OFF .. \
    && cmake --build . --target install \
    && cd ../.. \
    && rm -fr rabbitmq-c-${RABBITMQ_C_VERSION} \
    && rm v${RABBITMQ_C_VERSION}.tar.gz \
    && cd /usr/local/lib/ \
    && cp -r x86_64-linux-gnu/* . \
    && echo "===> clean up..." \
    && apt-get remove -q -y cmake \
    && apt-get clean -q -y \
    && apt-get autoremove -q -y \
    && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
```

Here are the library files:

```
# ls /usr/local/lib/librabbitmq.*
/usr/local/lib/librabbitmq.a   /usr/local/lib/librabbitmq.so.4
/usr/local/lib/librabbitmq.so  /usr/local/lib/librabbitmq.so.4.2.0
root@cd23d3589863:/python/daimajin# ls -l /usr/local/lib/librabbitmq.*
-rw-r--r-- 1 root staff 137652 Aug 10 00:18 /usr/local/lib/librabbitmq.a
lrwxrwxrwx 1 root staff     16 Aug 10 00:18 /usr/local/lib/librabbitmq.so -> librabbitmq.so.4
lrwxrwxrwx 1 root staff     20 Aug 10 00:18 /usr/local/lib/librabbitmq.so.4 -> librabbitmq.so.4.2.0
-rw-r--r-- 1 root staff  89872 Aug 10 00:18 /usr/local/lib/librabbitmq.so.4.2.0
```

**Comment by auvipy** [comment]
ENV RABBITMQ_C_VERSION=0.8.0?

**Comment by jimmywan** [comment]
I was just doing that as a convenience in my Dockerfile. AFAIK, it shouldn't affect anything.

**Comment by amitsaha** [comment]
@jimmywan you may want to check if `gcc` and `make` is installed?

**Comment by jimmywan** [comment]
@amitsaha The library appears to be building fine, hence the files in /usr/local/lib/librabbitmq.*.

Updated my original post a bit for clarity. rabbitmq-c appears to be building just fine, I just can't import librabbitmq.

**Comment by jimmywan** [comment]
Anyone still using this python library?

**Comment by multi-io** [comment]
I stumbled upon this problem when "make" wasn't installed -- in that case, the librabbitmq installation fails silently(!). librabbitmq includes the rabbitmq-c sources and builds and installs them (into the pip venv, i.e. usually into something like ../lib/python2.7/site-packages/_librabbitmq.so) during pip install librabbitmq -- if "make" is there, as I said.

**Comment by jimmywan** [comment]
@multi-io Thanks for the tip. Since I had built librabbitmq from source using the cmake based instructions I did not have make installed. I replaced cmake with gcc and make and now I can get everything running.

Unfortunately, some quick rudimentary testing using ApacheBench seems to indicate that using librabbitmq did not have a meaningful impact on application performance. Close to a dead heat in performance.
